cpp/include/cudf/reduction/detail/reduction.cuh

```diff
@@ -67,7 +67,16 @@ std::unique_ptr<scalar> reduce(InputIterator d_in,
 {
   auto const binary_op     = cudf::detail::cast_functor<OutputType>(op.get_binary_op());
   auto const initial_value = init.value_or(op.template get_identity<OutputType>());
-  auto dev_result          = rmm::device_scalar<OutputType>{initial_value, stream, mr};
+  auto host_scalar =
```

Comment:
Since this is just one value being copied to device, is there any performance advantage from first creating a pinned vector of size 1 and then using that to copy to device?